src/overlays/actors/ovl_En_Goma/z_en_goma.c

```diff
@@ -73,77 +104,795 @@ static ColliderCylinderInit D_80A4B7CC = {
+            this->actor.world.pos.z -= 1.5f;
+        }
+    }
+}
+
+void EnGoma_LookAtPlayer(EnGoma* this, GlobalContext* globalCtx) {
+    s16 eyePitch;
+    s16 eyeYaw;
+
+    eyeYaw = Actor_WorldYawTowardActor(&this->actor, &PLAYER->actor) - this->actor.shape.rot.y;
+    eyePitch = Actor_WorldPitchTowardActor(&this->actor, &PLAYER->actor) - this->actor.shape.rot.x;
+
+    if (eyeYaw > 6000) {
+        eyeYaw = 6000;
+    }
+    if (eyeYaw < -6000) {
+        eyeYaw = -6000;
+    }
+    Math_ApproachS(&this->eyeYaw, eyeYaw, 3, 2000);
+    Math_ApproachS(&this->eyePitch, eyePitch, 3, 2000);
+}
+
+void EnGoma_UpdateHit(EnGoma* this, GlobalContext* globalCtx) {
+    static Vec3f sShieldKnockbackVel = { 0.0f, 0.0f, 20.0f };
+    Player* player = PLAYER;
+
+    if (this->hurtTimer != 0) {
+        this->hurtTimer--;
+    } else {
+        ColliderInfo* acHitInfo;
+        u8 swordDamage;
+
+        if ((this->colCyl1.base.atFlags & 2) && this->actionFunc == EnGoma_Jump) {
+            EnGoma_SetupLand(this);
+            this->actor.speedXZ = 0.0f;
+            this->actor.velocity.y = 0.0f;
+        }
+
+        if ((this->colCyl2.base.acFlags & AC_HIT) && (s8)this->actor.colChkInfo.health > 0) {
+            acHitInfo = this->colCyl2.info.acHitInfo;
+            this->colCyl2.base.acFlags &= ~AC_HIT;
+
+            if (this->gomaType == ENGOMA_NORMAL) {
+                u32 dmgFlags = acHitInfo->toucher.dmgFlags;
+
+                if (dmgFlags & 0x100000) {
+                    if (this->actionFunc == EnGoma_Jump) {
+                        EnGoma_SetupLand(this);
+                        this->actor.velocity.y = 0.0f;
+                        this->actor.speedXZ = -5.0f;
+                    } else {
+                        Matrix_RotateY(player->actor.shape.rot.y / 32768.0f * 3.1415927f, MTXMODE_NEW);
+                        Matrix_MultVec3f(&sShieldKnockbackVel, &this->shieldKnockbackVel);
+                        this->invincibilityTimer = 5;
+                    }
+                } else if (dmgFlags & 1) { // stun
+                    if (this->actionFunc != EnGoma_Stunned) {
+                        EnGoma_SetupStunned(this, globalCtx);
+                        this->hurtTimer = 8;
+                    }
+                } else {
+                    swordDamage = CollisionCheck_GetSwordDamage(dmgFlags);
+
+                    if (swordDamage) {
+                        EffectSsSibuki_SpawnBurst(globalCtx, &this->actor.focus.pos);
+                    } else {
+                        swordDamage = 1;
+                    }
+
+                    this->actor.colChkInfo.health -= swordDamage;
+                    EnGoma_SetupHurt(this, globalCtx);
+                    Actor_SetColorFilter(&this->actor, 0x4000, 255, 0, 5);
+                    this->hurtTimer = 13;
+                }
+            } else {                           // die if still an egg
+                if (this->actor.params <= 5) { //! BossGoma only has 3 children
```

Comment:
think moving the comment down one line into the else block will play nicer with the formatter